circuit/src/execution-delivery/src/exec_composer/mod.rs

```diff
@@ -849,4 +892,335 @@ mod tests {
+            assert!(fetched_contract.is_ok());
+            // This assertion should logically pass but is failing. Doesnt makes sense.
+            // assert_eq!(fetched_contract.unwrap().code_hash, executable.code_hash);
+        });
+    }
+
+    #[test]
+    fn preload_bunch_of_contracts_multiple_contract_succeeds() {
+        let mut ext = TestExternalities::new_empty();
+        let requester =
+            AccountId32::from_str("5G9VdMwXvzza9pS8qE8ZHJk3CheHW9uucBn9ngW4C1gmmzpv").unwrap();
+        let value = BalanceOf::<Test>::from(0u32);
+        let temp_contract_one =
+            make_registry_contract_out_of_wat::<Test>(CODE_CALL, vec![], requester.clone(), value);
+        let temp_contract_two =
+            make_registry_contract_out_of_wat::<Test>(CODE_CALL, vec![], requester.clone(), value);
+        ext.execute_with(|| {
+            insert_default_xdns_record();
+
+            let res = ExecComposer::preload_bunch_of_contracts::<Test>(
+                vec![temp_contract_one, temp_contract_two],
+                Default::default(),
+            );
+
+            assert_ok!(res);
+        });
+    }
+
+    #[test]
+    fn preload_bunch_of_contracts_one_contract_fails() {
+        let mut ext = TestExternalities::new_empty();
+        let requester =
+            AccountId32::from_str("5G9VdMwXvzza9pS8qE8ZHJk3CheHW9uucBn9ngW4C1gmmzpv").unwrap();
+        let value = BalanceOf::<Test>::from(0u32);
+        let temp_contract_one = make_registry_contract_out_of_wat::<Test>(
+            INVALID_CODE,
+            vec![],
+            requester.clone(),
+            value,
+        );
+
+        ext.execute_with(|| {
+            insert_default_xdns_record();
+
+            let res = ExecComposer::preload_bunch_of_contracts::<Test>(
+                vec![temp_contract_one],
+                Default::default(),
+            );
+
+            assert_err!(res, "Can't decode WASM code");
+        });
+    }
+
+    #[test]
+    fn preload_bunch_of_contracts_multiple_contract_fails() {
+        let mut ext = TestExternalities::new_empty();
+        let requester =
+            AccountId32::from_str("5G9VdMwXvzza9pS8qE8ZHJk3CheHW9uucBn9ngW4C1gmmzpv").unwrap();
+        let value = BalanceOf::<Test>::from(0u32);
+        let temp_contract_one =
+            make_registry_contract_out_of_wat::<Test>(CODE_CALL, vec![], requester.clone(), value);
+        let temp_contract_two = make_registry_contract_out_of_wat::<Test>(
+            INVALID_CODE,
+            vec![],
+            requester.clone(),
+            value,
+        );
+
+        ext.execute_with(|| {
+            insert_default_xdns_record();
+            let res = ExecComposer::preload_bunch_of_contracts::<Test>(
+                vec![temp_contract_one, temp_contract_two],
+                Default::default(),
+            );
+
+            assert_err!(res, "Can't decode WASM code");
+        });
+    }
+
+    #[test]
+    fn run_single_contract_fails_stack_error() {
```

Comment:
Change naming to say when fails with stack error. Repeat in following test names too

Reply:
Hi @MaciejBaj , I have tried to improve the test names. Please take a look.
I have used this scheme : functionName_result_stateInfo wherever applicable.